runtime/gateway.go

```diff
@@ -111,9 +110,12 @@ type DefaultDependencies struct {
 	// Logger is a server-scoped logger
 	// Deprecated: Use ContextLogger instead.
 	Logger *zap.Logger
-
+	// ContextExtractor extracts context for scope and logs field
+	ContextExtractor ContextExtractor
```

Comment:
My understanding is it could be a singleton which init at gateway.go and pass it to where it needs like [router.go](https://github.com/uber/zanzibar/blob/0ee99879ee51fd34e9630729b355833585063af6/runtime/router.go#L167) and [tchannel_server.go](https://github.com/uber/zanzibar/blob/0ee99879ee51fd34e9630729b355833585063af6/runtime/tchannel_server.go#L138)

Reply:
OK. For now it is OK, we will refactor it in the future so that it's injected into `ContextMetrics` so it will be unnecessary.